lms/djangoapps/course_api/tests/test_views.py

```diff
@@ -206,6 +210,86 @@ def test_filter(self):
                 u"testing course_api.views.CourseListView with filter_={}".format(filter_),
             )
 
+    def test_filter_by_roles_global_staff(self):
+        """
+        Verify that global staff are always returned all courses irregardless of role filter.
+        """
+        self.setup_user(self.staff_user)
+
+        # Create a second course to be filtered out of queries.
+        alternate_course = self.create_course(org=md5(self.course.org).hexdigest())
+
+        # Request the courses as the staff user with the different roles specified.
+        for roles in ('', 'staff', 'staff,instructor'):
+            filtered_response = self.verify_response(params={'username': self.staff_user.username, 'role': roles})
```

Comment:
Did you mean for this to be `role` instead of `roles`?  Since you wanted to test with each given role?

Reply:
I meant to name it `roles` to get across the concept that zero, one, or more roles could be used as a filter on this API.

Reply:
Ah, got it. I misread the above line.  This is fine.